libsolidity/ast/Types.cpp

```diff
@@ -3090,6 +3098,25 @@ string FunctionType::canonicalName() const
 	return "function";
 }
 
+string FunctionType::humanReadableName() const
+{
+	string name;
+	switch (m_kind)
+	{
+	case Kind::Error:
+		name = "error " + m_declaration->name();
+		name += typesToString(m_parameterTypes, true);
```

Comment:
We already have a function you could use here:
```suggestion
		name = "error " + m_declaration->externalSignature();
```

I think that you should make `externalSignature()` use your new `typesToString()` though.

Reply:
After discussing with @ekpyron, `externalSignature()` is probably not what we want (it will print types like they are in the ABI) but what I really meant is that we definitely already have code for printing these parameters somewhere.

Maybe you could look inside `toString()` and check how it constructs the parameter list of a function. If there's a function for doing that - use it. If not, create such a function so that it can be reused.

Reply:
I looked at `FunctionType::externalSignature` but it seems to be different from what we have in `FunctionType::toString`, which was the original logic we are trying to reuse here, so I feel like we should differentiate these? Let me know what you think.

Reply:
Haha, sorry, while writing this I somehow missed the fact that the code I was looking at was actually inside one of the specializations of `toString()` :)

What I really meant its that I was sure we have code for printing of events/errors, including their signature implemented somewhere. But looks like I might be wrong - I just looked for it again and could not find anything. Apparently we really just print them as functions. In places that matter (like the output of `solc --hashes` we do use `externalSignature()`).

So ok, looks like this is the right way do it after all.